File: dotnet_backend/Services/AuthService.cs
using dotnet_backend.Repositories;
using dotnet_backend.Core;

namespace dotnet_backend.Services;

public class AuthService : IAuthService
{
    private readonly IUserRepository _userRepository;
    private readonly JwtService _jwtService;

    public AuthService(IUserRepository userRepository, JwtService jwtService)
    {
        _userRepository = userRepository;
        _jwtService = jwtService;
    }

    public string HandleSsoLogin(string provider, string subject, string? email)
    {
        _userRepository.Upsert(provider, subject, email);
        return _jwtService.CreateAccessToken(subject, provider, email);
    }
}

File: backend/app/services/auth_service.py
from sqlalchemy.orm import Session

from app.core.security import create_access_token
from app.repositories.user_repository import UserRepository


class AuthService:
    def __init__(self, user_repository: UserRepository):
        self._user_repository = user_repository

    def handle_sso_login(self, db: Session, *, provider: str, subject: str, email: str | None) -> str:
        self._user_repository.upsert(db, provider=provider, user_id=subject, email=email)
        db.commit()
        return create_access_token(subject=subject, provider=provider, email=email)
